src/test/java/org/broadinstitute/hellbender/engine/MultiVariantWalkerIntegrationTest.java

```diff
@@ -0,0 +1,157 @@
+    public void testQueryOverUnindexedFile() {
+        MultiVariantDataSource multiVariantSource = new MultiVariantDataSource();
+        multiVariantSource.addFeatureDataSource(new File(ENGINE_TEST_DIRECTORY + "unindexed.vcf"), "unindexed");
+        multiVariantSource.query(new SimpleInterval("1", 1, 1));
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testDuplicateSources() throws Exception {
+        final TestMultiVariantWalker tool = new TestMultiVariantWalker();
+
+        final List<String> args = new ArrayList<>();
+        File testFile = new File(getTestDataDir(), "baseVariants.vcf");
+        args.add("--variant");
+        args.add(testFile.getAbsolutePath());
+        args.add("--variant");
+        args.add(testFile.getAbsolutePath()); // add it again
+        tool.instanceMain(args.toArray(new String[args.size()]));
+    }
+
+    @DataProvider(name="variantFiles")
+    public Object[][] getVariantFiles() {
+        return new Object[][]
+            {
+                { Arrays.asList(new File(getTestDataDir(), "baseVariants.vcf")), null, 26 },
+                { Arrays.asList(new File(getTestDataDir(), "interleavedVariants_1.vcf")), null, 13 },
+                { Arrays.asList(
+                        new File(getTestDataDir(), "interleavedVariants_1.vcf"),
+                        new File(getTestDataDir(), "interleavedVariants_2.vcf")), null, 26 },
+                { Arrays.asList(
+                        new File(getTestDataDir(), "splitVariants_1.vcf"),
+                        new File(getTestDataDir(), "splitVariants_2.vcf")), null, 26 },
+
+                // with intervals
+                { Arrays.asList(new File(getTestDataDir(), "baseVariants.vcf")), "1", 14 },
+                { Arrays.asList(new File(getTestDataDir(), "interleavedVariants_1.vcf")), "1", 7 },
+                { Arrays.asList(
+                        new File(getTestDataDir(), "interleavedVariants_1.vcf"),
+                        new File(getTestDataDir(), "interleavedVariants_2.vcf")), "1", 14 },
+                { Arrays.asList(
+                        new File(getTestDataDir(), "interleavedVariants_1.vcf"),
+                        new File(getTestDataDir(), "interleavedVariants_2.vcf")), "2:200-600", 3 },
+            };
+    }
+
+    @Test(dataProvider = "variantFiles")
+    public void testVariantOrder(final List<File> inputFiles, final String interval, final int expectedCount) throws Exception {
+        final TestMultiVariantWalker tool = new TestMultiVariantWalker();
+
+        final List<String> args = new ArrayList<>(inputFiles.size() * 2);
+        inputFiles.forEach(f -> {args.add("--variant");
+        args.add(f.getAbsolutePath());});
+        if (interval != null) {
+            args.add("-L");
+            args.add(interval);
+        }
+
+        tool.instanceMain(args.toArray(new String[args.size()]));
+        Assert.assertEquals(tool.count, expectedCount);
+    }
+
```

Comment:
Add a test case in which multiple variants from different files are at the same position, and make sure the traversal can handle it.